internal/otel/otel.go

```diff
@@ -0,0 +1,125 @@
+package otel
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/metric/global"
+	"go.opentelemetry.io/otel/propagation"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.12.0"
+	"go.opentelemetry.io/otel/trace"
+	"google.golang.org/grpc"
+)
+
+// InitWithGRPC simplifies the creation of tracer and meter providers with GRPC
+func InitWithGRPC(
+	ctx context.Context, res *resource.Resource, target string, opts ...Option,
+) (
+	tp trace.TracerProvider,
+	mp metric.MeterProvider,
+	err error,
+) {
+	var c config
+	for _, opt := range opts {
+		opt(&c)
+	}
+
+	if !c.tracerProviderConfig.enabled && !c.meterProviderConfig.enabled {
+		return nil, nil, fmt.Errorf("no trace provider or meter provider to initialize")
+	}
+
+	conn, err := grpc.DialContext(ctx, target,
+		append(c.dialOpts, grpc.WithBlock())...,
+	)
+	if err != nil {
```

Comment:
on error, should we close the `conn`?

Reply:
https://github.com/grpc/grpc-go/blob/v1.51.0/clientconn.go#L160

Additionally, if you look at the rest of the function, when `err` is not `nil` then `conn` is `nil` so a `conn.Close` would give us a panic.

Reply:
here yes but after that we are using it right and error might occur there right then what happens to conn?

Reply:
Oh I get what you mean. I'll look into it, perhaps expose a `Shutdown` method :+1: